fetch-in-react-practice/client/src/components/Todos.js

```diff
@@ -0,0 +1,149 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import React, { useEffect, useState } from 'react';
+import PageTitle from './PageTitle';
+import TodoList from './TodoList';
+import TodoForm from './TodoForm';
+
+export default function Todos() {
+  const [todos, setTodos] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+
+  /* Implement useEffect to fetch all todos. Hints are at the bottom of the file. */
+  useEffect(() => {
+    async function getTodos() {
+      try {
+        const response = await fetch('/api/todos');
+        if (!response.ok) {
+          throw new Error(`Failed to fetch: ${response.status}`);
+        }
+        const currentTodoList = await response.json();
+        setTodos(currentTodoList);
+        setError(null);
+      } catch (error) {
+        setError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+
+    getTodos();
+  }, []);
+
+  /* Implement addTodo to add a new todo. Hints are at the bottom of the file. */
+  async function addTodo(newTodo) {
+    try {
+      const response = await fetch('/api/todos', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(newTodo),
+      });
+      if (!response.ok) {
+        throw new Error(`Failed to fetch: ${response.status}`);
+      }
+      const addedTodo = await response.json();
+      setTodos((prevTodos) => prevTodos.concat(addedTodo));
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  /* Implement toggleCompleted to toggle the completed state of a todo. Hints are at the bottom of the file. */
+  async function toggleCompleted(todoId) {
+    try {
+      const todoIndex = todos.findIndex((todo) => todo.todoId === todoId);
+
+      const todoToUpdate = todos[todoIndex];
+      const updatedTodo = {
+        ...todoToUpdate,
+        isCompleted: !todoToUpdate.isCompleted,
+      };
```

Comment:
The PATCH implementation in the server just needs `isCompleted`. It does not need the rest of the Todo.